src/components/YearlyDataTab.tsx

```diff
@@ -88,13 +89,13 @@ export default class YearlyDataTab extends React.Component<YearlyDataTabProps, Y
       return (
         <li key={y} className="year-li">
           <section className={this.state.styled ? "header-bar" : ""}>
-            <span>{y}: {yearlyTotal} librar{yearlyTotal !== 1 ? "ies" : "y"} added</span>
+            <span>{y}: {yearlyTotal} librar{yearlyTotal !== 1 ? "ies" : "y"} validated</span>
             <span>({getPercentage(yearlyTotal, total, true)})</span>
           </section>
           {
             this.state.yearsToShow[y] &&
             <section className={this.state.styled ? "list-holder" : ""}>
-            { <StatsInnerList data={sortedByYear[y]} styled={this.state.styled} showMonths={this.state.months} /> }
+            { <StatsInnerList data={sortedByYear[y]} hasYear={y !== "Unknown"} styled={this.state.styled} showMonths={this.state.months} /> }
```

Comment:
You do similar checks in this component and in `StatsInnerList`:
```
let validated = library.urls_and_contact.contact_validated;
let year = validated === "Not validated" ? ....
```
Since you're passing `hasYear` here, maybe you can pass the validation as well so you don't have to do it again in the child component. Not necessary to do but just a thought

Reply:
After looking at all the files, there's the `validate` function from the TestUtils but maybe you can create a similar function that does something similar with real data rather than with the test data. That way you can update the library's object once instead of checking and updating values in two different components.